### ISSUE TITLE
Copy Attributes From One Layer to Another for Single or Multiple Fields Based on Join

### ISSUE THREAD

**Issue by ar-siddiqui** [qgis/QGIS#42505]
**Feature description.**
I have two layers A & B, I want to copy fields from Layers B to Layers A based on a join. Currently, I can do this joining A&B, and then one by one copying fields using the field calculator for each field. It would be nice to an algorithm that can do that for me.

**Additional context**
The Algorithem can ask for input and target layer > ask for join field in each > ask which fields to copy from B to A and with what matching criteria, for example, a user can say field B_1 from layer B should be copied to A_2 from layer A and so on

**Comment by roya0045** [comment]
I'm pretty sure that saving the layer with the join also include the joined information.

**Comment by ar-siddiqui** [comment]
> I'm pretty sure that saving the layer with the join also include the joined information.

That is correct, but this feature request is different. I have run across cases multiple times where you need to copy attributes from joined table to your actual layer, not save them as added attributes.

For example, you have the world_cities layer from 2018 and you got new world_cities statistics from 2020 like population, area, etc in CSV format. Now you want to update your 2018 file to 2020 and you want to copy over attributes from the CSV file. The CSV file may or may not have the same field names. It looks to me that this will be a useful feature that many people may utilize.

**Comment by ar-siddiqui** [comment]
This is how I am doing it through pyQGIS ... thought this might be useful

```python
### Inputs

# replace names here in quotation marks 
copy_from = "xsecs_results" 
copy_to = "xsecs" 
join_field = "Section"
field_to_copy = "Manning_n" 


### Processing 

xsecs_results = QgsProject.instance().mapLayersByName(copy_from)[0]
xsecs = QgsProject.instance().mapLayersByName(copy_to)[0]

joinObject = QgsVectorLayerJoinInfo()
joinObject.setJoinLayer(xsecs_results)
joinObject.setJoinFieldName(join_field)
joinObject.setTargetFieldName(join_field) 
joinObject.setUsingMemoryCache(True)
joinBuffer = xsecs.addJoin(joinObject)

#fields = xsecs.fields():
#
#source_field_id = fields.indexFromName(field_to_copy)
#target_field_id = fields.indexFromName(copy_from+"_"+field_to_copy)
expr_string = f'"{copy_from}_{field_to_copy}" is not NULL'
expr = QgsExpression(expr_string)
features = xsecs.getFeatures(QgsFeatureRequest(expr))

xsecs.startEditing()
for feat in features:
    feat[field_to_copy] = feat[f'{copy_from}_{field_to_copy}']
    xsecs.updateFeature(feat)
```

**Comment by edge540T** [comment]
> This is how I am doing it through pyQGIS ... thought this might be useful
> 
> ```python
> ### Inputs
From 0 on python to do what I was need it, thanks. There is a way to do the same coy paste but in windows 1254 instead utf-8? Some previous command I mean, may be I am saying something wrong... Thanks anyway again.
